### PR TITLE
feat(web/lifecycleService): correct startupKind

### LINKED ISSUE #206345: LifecycleService shutdownReason should read `performance.navigation.type`
In the `LifecycleService` you have a `startupKind` that on electron is used in stuff like the layoutService to determine whether to restore the sidebar contents only on a reload. But on web this is hardcoded to always restore the contents, not just on a reload.

It's possible to detect the navigation type by doing `performance.navigation.type === 1` which would indicate a reload. So I suggest you should read the property, and expose it to under the `startupKind` property (removing all the hardcoding on the web) so all the features work as expected

### PR DESCRIPTION
**Pull request by samdenty**
This PR fixes #206345 & #206296. 

Before the `startupKind` was only implemented for electron, but we can also do it on web (only found out cause chatgpt suggested). This makes the editor restoration logic better (and i'm sure other parts that use startupKind too)

-----
<a href="https://stackblitz.com/~/github.com/samdenty/vscode/tree/unexpected-mouse">Review PR in StackBlitz</a> _Submitted with [StackBlitz](https://stackblitz.com/~/github.com/samdenty/vscode/tree/unexpected-mouse)._

### CODE REVIEW

File: src/vs/workbench/browser/layout.ts [hunk 1]
```diff
@@ -671,7 +671,7 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 
 			// Only restore last viewlet if window was reloaded or we are in development mode
 			let viewContainerToRestore: string | undefined;
-			if (!this.environmentService.isBuilt || lifecycleService.startupKind === StartupKind.ReloadedWindow || isWeb) {
+			if (!this.environmentService.isBuilt || lifecycleService.startupKind === StartupKind.ReloadedWindow) {
```
Comment (bpasero) [inline src/vs/workbench/browser/layout.ts:674]: How is this related?

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/layout.ts (hunk 1 above)
Comment (samdenty) [reply src/vs/workbench/browser/layout.ts:674]: because web didn't support startupKind reloaded window before now, it was hardcoded to always restore files on web. But now it's supported, we can use the dynamic behaviour the electron version has

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/layout.ts [hunk 2]
```diff
@@ -2611,7 +2611,7 @@ class LayoutStateModel extends Disposable {
 		LayoutStateKeys.GRID_SIZE.defaultValue = { height: workbenchDimensions.height, width: workbenchDimensions.width };
 		LayoutStateKeys.SIDEBAR_SIZE.defaultValue = Math.min(300, workbenchDimensions.width / 4);
 		LayoutStateKeys.AUXILIARYBAR_SIZE.defaultValue = Math.min(300, workbenchDimensions.width / 4);
-		LayoutStateKeys.PANEL_SIZE.defaultValue = (this.stateCache.get(LayoutStateKeys.PANEL_POSITION.name) ?? LayoutStateKeys.PANEL_POSITION.defaultValue) === 'bottom' ? workbenchDimensions.height / 3 : workbenchDimensions.width / 4;
+		LayoutStateKeys.PANEL_SIZE.defaultValue = (this.stateCache.get(LayoutStateKeys.PANEL_POSITION.name) ?? LayoutStateKeys.PANEL_POSITION.defaultValue) === Position.BOTTOM ? workbenchDimensions.height / 3 : workbenchDimensions.width / 4;
```
[the remark this answers is not in the document]
Comment (samdenty) [reply src/vs/workbench/browser/layout.ts:2614]: in our fork, we open the terminal by default and I noticed it had the wrong height when i launched with devtools open

**Review by bpasero — APPROVED** [review]

**Comment by bpasero** [comment]
@samdenty any thoughts on https://web.dev/articles/navigation-and-resource-timing#acquiring_timings_in_application_code. I wonder about the actual perf impact of calling `performance.getEntriesByType('navigation')`. In my testing on macOS, I could not find this call to take long, but I wanted to point out the blog post that suggests a observer pattern.

**Comment by samdenty** [comment]
> iterating over a large number of entries, or even repeatedly polling the performance buffer to find new entries

I think since this we're calling `performance.getEntriesByType('navigation')` there should only ever be one event, so browsers shouldn't have a hard time resolving that

**Comment by bpasero** [comment]
👍 , I think this can go in, I pushed some slight 👄 changes.

One thing that will be received as some kind of regression potentially is that in web, we no longer just restore the previous view you had opened (you removed the `isWeb` check). But on the other hand, that is consistent with the behaviour on desktop and as such actually the correct behaviour. I hope this will not backfire though...

**Comment by bpasero** [comment]
Interesting: https://github.com/microsoft/vscode/commit/5727c820a26ec2e272c1e18e347768c36e9845ea

**Review by mjbvz — APPROVED** [review]
